screens/LocationScreen.js

```diff
@@ -0,0 +1,201 @@
+import React, { useState, useEffect } from 'react';
+import { StyleSheet, TextInput, TouchableOpacity } from 'react-native';
+import { connect } from 'react-redux'
+import { WebViewLeaflet } from 'react-native-webview-leaflet'
+
+import { Actions } from "react-native-router-flux";
+import { Container, Content, Header, View } from "../components/layout";
+import { Button } from "../components/buttons";
+import { Form, Input } from "../components/forms"
+import { Text } from "../components/typography"
+import theme from '../styles/base'
+import { getCurrentLocation, convertToCoordinates, convertToAddress } from '../utils/index'
+import { getUserLocation, updateUserLocation } from '../store/actions';
+
+import mapMarkerIcon from '../components/icons/mapMarker'
+
+const INITIAL_COORDINATES = {
+     // Default location is set to Vancouver
+     lat: 49.2827,
+     lng: -123.1207
+}
+
+const TEMPLATE_ADDRESSES = [
+    '4070 West 38th Ave, Vancouver',
+    '100 Universal City Plaza, Universal City, California'
+]
+
+const openStreetMapLayer = {
+    attribution:'&amp;copy <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
+    baseLayerIsChecked: true,
+    baseLayerName: "OpenStreetMap.Mapnik",
+    url: "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+}
+
+const LocationScreen = (props) => {
+    const [webViewLeafletRef, setWebViewLeafletRef] = useState(null)
+    const [mapCenterPosition, setMapCenterPosition] = useState(INITIAL_COORDINATES)
+    const [mapZoom, setMapZoom] = useState(7)
+    const [mapLayers, setMapLayers] = useState(null)
+    const [mapMarkers, setMapMarkers] = useState([])
+
+    const [address, setAddress] = useState(null)
+    const [addressConfirm, setAddressConfirm] = useState(false) 
```

Comment:
Since we would need the address when a help request is sent out to the responders, there are a couple options:
- Save this in a redux so it can be easily retrieved
- Let backend convert the lat lon, use same library etc.
- Save the address text itself instead (because it seems to be quite reliable from my testing)

Which one do you think should be the best for this scenario?

Reply:
Agree with having it stored somewhere for convenience and to cut down on unnecessary requests. I think the most reliable option is to save the coordinates into redux for now

Reply:
Yes go ahead with that, sorry for replying two days late 🙏